app/src/androidTest/java/ca/dal/cs/athletemonitor/athletemonitor/AccountManagerTest.java

```diff
@@ -0,0 +1,315 @@
+//        AccountManager.createUser(createTestUser());
+//        AccountManager.authenticate(testUser, assertTrueBooleanResult());
+//
+//        //attempt to sign out
+//        AccountManager.signOut(testUser, assertTrueBooleanResult());
+//
+//        //delete test user from database
+//        AccountManager.deleteUser(testUser, assertTrueBooleanResult());
+//    }
+
+    /**
+     * Tests creation of a new user
+     *
+     * Will pass if new account is added to the users branch of database
+     *
+     * @throws Exception Generic exception
+     */
+    @Test
+    public void createUserTest() throws Exception {
+       final User testUser = TestingHelper.createTestUser();
+
+       AccountManager.createUser(testUser, new AccountManager.UserObjectListener() {
+           @Override
+           public void onUserPopulated(User user) {
+               assertNotNull(user);
+
+               //if the user was created successfully, remove before exiting
+               AccountManager.deleteUser(testUser, TestingHelper.assertTrueBooleanResult());
+           }
+       });
+    }
+
+    /**
+     * Test for true positive on existing user
+     *
+     * Will pass in a username that does exist in the user accounts to the validate method and
+     * assert that the return value of validate is true
+     *
+     * @throws Exception Generic exception
+     */
+    @Test
+    public void userExistsTest() throws Exception {
+        final User testUser = TestingHelper.createTestUser();
+        AccountManager.createUser(TestingHelper.createTestUser());
+
+        AccountManager.userExists(testUser.getUsername(), new AccountManager.UserExistsListener() {
+            @Override
+            public void onUserExists(boolean result) {
+                assertTrue(result);
+
+                //delete the test user before exiting test
+                AccountManager.deleteUser(testUser, null);
+            }
+        });
+    }
+
+    /**
+     * Test for false positive on existing user
+     *
+     * Will pass if a username that does not exist in the user accounts is passed to the userExists
+     * method.  Asserts that the return value of is false
+     *
+     * @throws Exception Generic exception
+     */
+    @Test
+    public void userDoesNotExistTest() throws Exception {
+        User testUser = new User("i don't exist", "testPassword");
+
+        AccountManager.userExists(testUser.getUsername(), new AccountManager.UserExistsListener() {
+            @Override
+            public void onUserExists(boolean result) {
+                assertFalse("User should not exist, but user is found.", result);
+            }
+        });
+    }
+
+    /**
+     * Test for deleting a user account.
+     *
+     * First checks if the test user exists in the database.  If it does not exist, it is created
+     * and then a deleteUser call is made.  If the delete operation is completed without errors
+     * the test is assumed to be passed.
```

Comment:
Is it possible to actually check that the user no longer exists in the database? For example, what if the delete operation failed to delete the user, but didn't throw an exception so this test still passes.

Reply:
Yes, it is possible to check by instantiating a DatabaseReference to the node that was deleted.  This will result in false being returned on the DataSnapshot paramater of the ValueEventListener.  

I'm not sure if it is necessary to check though since the callback for DatabaseReference.removeValue() will be called on both success and failure.  If the DatabaseError object passed to the callback is null, the operation was successful.  This result is returned to the deleteUser listener.